devices/HapticGlove/src/HapticGlove.cpp

```diff
@@ -537,22 +552,29 @@ class HapticGlove::SenseGloveImpl::SenseGloveHapticActuator : public wearable::a
 
     bool setHapticCommand(double& value) const override
     {
+        yError() << LogPrefix << "Wrong method has been called! To set the haptic command please use the setHapticsCommand method.";
+        return false;
+    }
 
-        std::lock_guard<std::mutex> lock(m_gloveImpl->mutex);
+    bool setHapticCommands(std::vector<double>& forceValue, std::vector<double>& vibrotactileValue) const override
+    {
 
-        if (!m_gloveImpl->isAvailable(m_actuatorName,
-                                      m_gloveImpl->gloveData.humanHapticActuatorNameIdMap)) {
-            yError() << LogPrefix << "The actuator name (" << m_actuatorName
-                     << ") is not found in the list of actuators.";
+        std::lock_guard<std::mutex> lock(m_gloveImpl->mutex);
 
+        for (size_t i = 0; i < m_gloveImpl->nFingers; i++)
+        {
+            m_gloveImpl->gloveData.fingersHapticFeedback[i] = forceValue[i];
+            m_gloveImpl->gloveData.fingersHapticFeedback[i + 5] = vibrotactileValue[i];
+        }
+        if (forceValue.size() != m_gloveImpl->nActuatorsPerGlove || vibrotactileValue.size() != m_gloveImpl->nActuatorsPerGlove)
+        {
+            yError() << "The sizes of the forceValue and the vibrotactileValue vectors are not correct!";
```

Comment:
This should be checked before accessing the vectors, otherwise you risk a segmentation fault